### PR TITLE
Use `to_form` helper for forms and refactor inputs

### PR DESCRIPTION
**Pull request by Flo0807**
- use `to_form` helper to create forms
- refactor inputs to not rely on `phoenix_html` helper functions

### CODE REVIEW

File: lib/backpex/html/form.ex [hunk 1]
```diff
@@ -4,152 +4,249 @@ defmodule Backpex.HTML.Form do
...
+
+  slot :inner_block
+
+  def field_input(%{field: %Phoenix.HTML.FormField{} = field} = assigns) do
+    translated_errors =
+      field.errors
+      |> Enum.map(fn error ->
+        {text, _error} = translate_form_error(error, assigns.field_options)
+        text
+      end)
```
Comment (pehbehbeh) [inline lib/backpex/html/form.ex:53]: We also have this code in `multi_select`. Maybe add a `translated_form_errors` function?

File: lib/backpex/html/form.ex [hunk 2]
```diff
@@ -4,152 +4,249 @@ defmodule Backpex.HTML.Form do
...
+            id={@id}
+            name={@name}
+            value="true"
+            checked={@checked}
+            class={[
+              "toggle phx-no-feedback:toggle phx-no-feedback:toggle-primary",
+              @input_class,
+              @errors == [] && "toggle-primary",
+              @errors != [] && "toggle-error"
+            ]}
```
Comment (pehbehbeh) [inline lib/backpex/html/form.ex:149]: Maybe add a helper for this common case?

- first param: default class
- second param: no error classes
- third param: error classes

```suggestion
            class={input_classes(["toggle phx-no-feedback:toggle phx-no-feedback:toggle-primary", @input_class], "toggle-primary", "toggle-error")}
```

File: lib/backpex/html/resource.ex [hunk 3]
```diff
@@ -961,13 +960,22 @@ defmodule Backpex.HTML.Resource do
   defp metric_toggle(assigns) do
     visible = Backpex.Metric.metrics_visible?(assigns.metric_visibility, assigns.live_resource)
 
-    assigns = assign(assigns, :visible, visible)
+    form =
+      to_form(
+        %{"_resource" => assigns.live_resource, "_cookie_redirect_url" => assigns.current_url},
+        as: :toggle_metrics
+      )
```
Comment (pehbehbeh) [inline lib/backpex/html/resource.ex:967]:
```suggestion
    form =
      %{"_resource" => assigns.live_resource, "_cookie_redirect_url" => assigns.current_url}
      |> to_form(as: :toggle_metrics)
```

**Review by pehbehbeh — APPROVED** [review]
Awesome refactoring :) I only noted a few small things. You can merge on your own.

File: lib/backpex/html/form.ex (hunk 2 above)
Comment (Flo0807) [reply lib/backpex/html/form.ex:149]: In this case, would we also need to pass the `@errors` attribute to the `input_classes` function, or how does it decide whether to include error classes or no error classes?

Comment (pehbehbeh) [reply lib/backpex/html/form.ex:149]: Right... maybe just leave it as is.

**Pull request merged**